### DISCUSSION TITLE
single line separation in comment

### DISCUSSION THREAD

**Discussion by lukgru** [PHP-CS-Fixer/PHP-CS-Fixer#8251] [Q&A]
Hi,
in the original version I have something like this

    /**
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

After use rule sets @Symfony i have something like this

    /**
     * @ORM\Column(name="id", type="integer")
     * 
     * @ORM\Id
     * 
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

Is there any way to go back to the original version and remove these blank lines?

**Comment by Raymond-Ly** [comment]
Hi, I believe this is the rule you are looking for: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/phpdoc/phpdoc_separation.rst

E.g. in `.php-cs-fixer.php`, you can specify

```
return PhpCsFixer\Config::create()
    ->setRules([
        'phpdoc_separation' => false, // Disables blank lines between annotations
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->in(__DIR__)
    );

```

**Comment by salimuddin07** [comment]
Option 1: Disable or customize the PHPDoc separation rule

You can create a .php-cs-fixer.dist.php (or edit your existing config) and disable or tweak the rule:

```
<?php

$finder = PhpCsFixer\Finder::create()
    ->in(__DIR__.'/src'); // your source directory

return (new PhpCsFixer\Config())
    ->setRules([
        '@Symfony' => true,
        'phpdoc_separation' => false, // disable extra blank lines between PHPDoc tags
    ])
    ->setFinder($finder);
```


phpdoc_separation is responsible for inserting blank lines between annotations.

Setting it to false keeps your original style (no extra blank lines).

After updating the config, run PHP CS Fixer again:

`php-cs-fixer fix`


This should restore your PHPDoc blocks to the original format.

Option 2: Use a more specific ruleset

If you want most @Symfony formatting but without PHPDoc blank lines, you can combine the rulesets:

```
return (new PhpCsFixer\Config())
    ->setRules([
        '@Symfony' => true,
        'phpdoc_separation' => false,
        // any other rules you want to override
    ])
```


This way, you retain other Symfony coding standards while controlling the PHPDoc formatting.

Option 3: Manually revert (not recommended for many files)

You could manually remove blank lines in your PHPDoc blocks.

But for large projects or ongoing formatting, adjusting the fixer configuration is the only scalable solution.

**Comment by kubawerlos** [comment]
https://cs.symfony.com/doc/rules/phpdoc/phpdoc_separation.html#example-4 - you can configure the rule to your needs:
```
$ cat test.php 
<?php class Entity {
    /**
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     *
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;
}

$ php php-cs-fixer check -vvv --diff test.php --rules=phpdoc_separation
Paths from configuration file have been overridden by paths provided as command arguments.
   1) test.php (phpdoc_separation)
      ---------- begin diff ----------
--- /home/kuba/code/friendsofphp/PHP-CS-Fixer/test.php
+++ /home/kuba/code/friendsofphp/PHP-CS-Fixer/test.php
@@ -1,6 +1,7 @@
 <?php class Entity {
     /**
      * @ORM\Column(name="id", type="integer")
+     *
      * @ORM\Id
      *
      * @ORM\GeneratedValue(strategy="AUTO")

      ----------- end diff -----------

$ php php-cs-fixer check -vvv --diff test.php --rules='{"phpdoc_separation":{"groups":[["ORM\\*"]]}}'
Paths from configuration file have been overridden by paths provided as command arguments.
   1) test.php (phpdoc_separation)
      ---------- begin diff ----------
--- /home/kuba/code/friendsofphp/PHP-CS-Fixer/test.php
+++ /home/kuba/code/friendsofphp/PHP-CS-Fixer/test.php
@@ -2,7 +2,6 @@
     /**
      * @ORM\Column(name="id", type="integer")
      * @ORM\Id
-     *
      * @ORM\GeneratedValue(strategy="AUTO")
      */
     protected $id;

      ----------- end diff -----------


```
